project_management_system/src/app/auth/signup/signup.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+import { Router, ActivatedRoute } from '@angular/router';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { first } from 'rxjs/operators';
+import { TranslateService } from '@ngx-translate/core';
+
+import { AccountService } from '../../services/account.service';
+import { AlertService } from '../../services/alert.service';
+
+@Component({
+  selector: 'app-signup',
+  templateUrl: './signup.component.html',
+  styleUrls: ['./signup.component.scss'],
+})
+export class SignupComponent implements OnInit {
+  form!: FormGroup;
+  loading = false;
+  submitted = false;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private route: ActivatedRoute,
+    private router: Router,
+    private accountService: AccountService,
+    private alertService: AlertService,
+    private translate: TranslateService
+  ) {}
+
+  ngOnInit() {
+    this.form = this.formBuilder.group({
+      name: ['', Validators.required],
+      login: ['', Validators.required],
+      password: ['', [Validators.required, Validators.minLength(6)]],
+    });
+    this.translate.onLangChange.subscribe();
```

Comment:
Every time visiting this page will be created a new subscription. This leads to performance issue. So you should pay attention to closing them. It might be unsubscribing ( in the onDestroy method) or approaches from the article https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f. In the projects' life, we use take* more often than others.